src/tests/workflows/recipes/components/FilterObjectForm.test.js

```diff
@@ -0,0 +1,509 @@
+          const value = filterObject.get(key.split('.')[1]);
+          if (value) {
+            return value.toJS();
+          }
+        },
+        getFieldDecorator: (key, config) => {
+          return Component => {
+            return;
+          };
+        },
+      },
+      filterObject,
+      onSubmit: jest.fn(),
+    };
+
+    const FakeForm = createForm({})(FilterObjectForm);
+    const { getByText } = render(wrapMockStore(<FakeForm {...props} />));
+    expect(getByText('Geo')).toBeTruthy();
+    expect(getByText('Browser')).toBeTruthy();
+    expect(getByText('Sampling')).toBeTruthy();
+    expect(fetchMock.called()).toBeTruthy();
+  });
+
+  it('render editing form with a full existing filter object', () => {
+    fetchMock.getOnce(`${NORMANDY_ADMIN_API_ROOT_URL}v3/filters/`, {
+      channels: [
+        {
+          key: 'beta',
+          value: 'Beta',
+        },
+      ],
+      countries: [{ key: 'SE', value: 'Sweden' }],
+      locales: [{ key: 'sv', value: 'Swedish' }],
+    });
+
+    const filterObject = serializeFilterObjectToMap(
+      fromJS([
+        { type: 'channel', channels: ['beta'] },
+        { type: 'country', countries: ['SE'] },
+        { type: 'locale', locales: ['sv'] },
+        { type: 'channel', channels: ['beta'] },
+        { type: 'version', versions: [67, 68, 69] },
+        { type: 'bucketSample', start: 10, count: 20, total: 30, input: [] },
+      ]),
+    );
+    const props = {
+      disabled: false,
+      form: {
+        getFieldValue: key => {
+          // Remember, 'key' is something like "filter_object.locales"
+          const value = filterObject.get(key.split('.')[1]);
+          if (value) {
+            return value.toJS();
+          }
+        },
+        getFieldDecorator: (key, config) => {
+          return Component => {
+            return;
+          };
+        },
+      },
+      allChannels: fromJS([{ value: 'Release', key: 'release' }]),
+      allCountries: fromJS([{ value: 'Sweden', key: 'SE' }]),
+      allLocales: fromJS([{ value: 'Swedish', key: 'sv' }]),
+      filterObject,
+      onSubmit: jest.fn(),
+    };
+
+    const FakeForm = createForm({})(FilterObjectForm);
+    const { getByText } = render(wrapMockStore(<FakeForm {...props} />));
+
+    // Expect to see "Geo (2)" because a country and a locale is chosen.
+    expect(getByText('Geo (2)')).toBeTruthy();
+    // Same with "Browser (2)" because both channels and versions is chosen.
+    expect(getByText('Browser (2)')).toBeTruthy();
+    // And "Sampling (1)" because a sampling method has been chosen.
+    expect(getByText('Sampling (1)')).toBeTruthy();
+
+    expect(fetchMock.called()).toBeTruthy();
+  });
+
+  describe('<serializeFilterObjectToMap>', () => {
+    it('serialize an empty List to an empty Map', () => {
+      const map = serializeFilterObjectToMap(List());
+      expect(map).toBeImmutableMap();
+      expect(map.size).toEqual(0);
```

Comment:
It's probably easier to say `expect(map).toEqualImmutable(Map())`

Reply:
Cool. I'll take it.